### ISSUE TITLE
[Feature Request] Flow screenshots record before every action

### ISSUE THREAD

**Issue by OHoussein** [mobile-dev-inc/maestro#1526]
**Problem**
When you run a full flow, it's not easy to understand what steps has been done and how the app reacted on each action

**Solution**
Generate automatically a screenshot before every action like tapOn, back (can be added through and option) and generate a report after the test execution, it will look like this screen below
![Capture d’écran 2023-10-13 à 14 53 26](https://github.com/mobile-dev-inc/maestro/assets/10960959/b46199dd-441d-4f4e-8d47-e48a3d389320)

**Comment by OHoussein** [comment]
I'm open to contribute to the project to add this feature :)

**Comment by bartekpacia** [comment]
Yeah, `takeScreenshot` should solve this, but I understand that the request here is to essentially have `takeScreenshot` called automatically before each action?

**Comment by OHoussein** [comment]
Exactly, doing takeScreenshot manually after each action doesn't look productive. 
The goal of this is to have a quick understanding of the test execution.
